### PR TITLE
Moved the scan of each individual picture in an BackgroundJob 

### PR DESCRIPTION
**Pull request by tacruc**
This will make it restore if it crashes on a file.
This will reduce the impact of  #126 and #206.

### CODE REVIEW

File: lib/BackgroundJob/AddPhotoJob.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                                IRootFolder $root,
+                                PhotofilesService $photofilesService) {
+        parent::__construct($timeFactory);
+        $this->photofilesService = $photofilesService;
+        $this->root = $root;
+    }
+
+    public function run($arguments) {
+        $userFolder = $this->root->getUserFolder($arguments['userId']);
+        $file = $userFolder->getById($arguments['photoId'])[0];
```
Comment (nickvergessen) [inline lib/BackgroundJob/AddPhotoJob.php]:
```suggestion
        $files = $userFolder->getById($arguments['photoId']);
        if (empty($files)) {
                return;
        }
        $file = array_shift($files);
```

File: lib/BackgroundJob/UpdatePhotoByFileJob.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+class UpdatePhotoByFileJob extends QueuedJob {
+
+    private $jobList;
+
+    /**
+     * UserInstallScanJob constructor.
+     *
+     * A QueuedJob to scan user storage for photos and tracks
+     *
+     * @param IJobList $jobList
```
Comment (nickvergessen) [inline lib/BackgroundJob/UpdatePhotoByFileJob.php]: invalid doc block

File: lib/BackgroundJob/UpdatePhotoByFileJob.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+     */
+    public function __construct(ITimeFactory $timeFactory,
+                                PhotofilesService $photofilesService) {
+        parent::__construct($timeFactory);
+        $this->photofilesService = $photofilesService;
+    }
+
+    public function run($arguments) {
+        $userFolder = $this->root->getUserFolder($arguments['userId']);
+        $file = $userFolder->getById($arguments['fileId'])[0];
```
Comment (nickvergessen) [inline lib/BackgroundJob/UpdatePhotoByFileJob.php]: see above

File: lib/BackgroundJob/UpdatePhotoByFileJob.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+     */
+    public function __construct(ITimeFactory $timeFactory,
+                                PhotofilesService $photofilesService) {
+        parent::__construct($timeFactory);
+        $this->photofilesService = $photofilesService;
+    }
+
+    public function run($arguments) {
+        $userFolder = $this->root->getUserFolder($arguments['userId']);
+		$files = $userFolder->getById($arguments['photoId']);
```
Comment (nickvergessen) [inline lib/BackgroundJob/UpdatePhotoByFileJob.php]:
```suggestion
		$files = $userFolder->getById($arguments['fileId']);
```

File: lib/BackgroundJob/UpdatePhotoByFileJob.php [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+		if (empty($files)) {
+			return;
+		}
+		$file = array_shift($files);
+        if ($file !== null)
+            $this->photofilesService->updateByFileNow($file);
```
Comment (nickvergessen) [inline lib/BackgroundJob/UpdatePhotoByFileJob.php]: Should wrap with parentheses

File: lib/BackgroundJob/AddPhotoJob.php [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+    public function run($arguments) {
+        $userFolder = $this->root->getUserFolder($arguments['userId']);
+        $files = $userFolder->getById($arguments['photoId']);
+        if (empty($files)) {
+                return;
+        }
+        $file = array_shift($files);
+        if ($file !== null)
+            $this->photofilesService->addPhotoNow($file, $arguments['userId']);
```
Comment (nickvergessen) [inline lib/BackgroundJob/AddPhotoJob.php]: parentheses

**Comment by mwlczk** [comment]
background scan of encrypted files does not work. 
Maps does not work anymore for encrypted installations :(

**Pull request merged**